### PR TITLE
Revamp `authorId` string field into `authorIds` list field

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

**Review by szymon-rd — APPROVED** [review]

**Comment by propensive** [comment]
Is this safe to merge, given that the backend presumably needs to know about the changes?

**Comment by mkondratek** [comment]
> Is this safe to merge, given that the backend presumably needs to know about the changes?

Not yet :) I have to ensure that everything works fine in dev environment first

File: topics/collections/index.json [hunk 1]
```diff
@@ -5,7 +5,10 @@
     {
       "id": "basics",
       "title": "Basics",
-      "authorId": "szymon-rodziewicz",
+      "authorIds": [
```
Comment (mkondratek) [inline topics/collections/index.json:8]: reorder here and everywhere else

**Comment by propensive** [comment]
I've noticed this has been open for a while... as the `dev` and `main` branches are starting to get a bit out-of-sync.

I guess there's some nervousness about making the change when it could cause the data to be out of sync with the rest of the site for some time. But as a first step, could we include both `authorId` and `authorIds` (with all the duplication that entails) so that the JSON would contain all the data necessary for either version? And then once everything is upgraded and working, remove the old `authorId` fields?

**Comment by mkondratek** [comment]
> I've noticed this has been open for a while... as the `dev` and `main` branches are starting to get a bit out-of-sync.
> 
> I guess there's some nervousness about making the change when it could cause the data to be out of sync with the rest of the site for some time. But as a first step, could we include both `authorId` and `authorIds` (with all the duplication that entails) so that the JSON would contain all the data necessary for either version? And then once everything is upgraded and working, remove the old `authorId` fields?

Good point about `authorId(s)` but since there is a lot of changes (authors, moved topics, moved images, restructure)...
@kusmierz suggested that the best solution would be removing all deprecated / "temporary" code and fields (like `authorId` in particular) and simply reconcile with some downtime :)

**Comment by propensive** [comment]
That's probably fine too. I guess it won't be a _lot_ of downtime. I've stopped using `dev` for testing new lessons the last few weeks due to the differences in the format of the content, so I'm pushing straight to the `main` content branch, and then checking it. All this means from your point of view is that the `main` branch should be treated as *definitive* when you're doing the migration. That's probably going to be most relevant when reconciling the JSON.

**Pull request merged**